tests/test_attempt_user_login.py

```diff
@@ -0,0 +1,93 @@
+# pylint: disable=no-self-use
+#       - pylint test classes must pass self, even if unused.
+# pylint: disable=logging-fstring-interpolation
+#       - honestly just annoying to use lazy(%) interpolation.
+"""
+Endpoint testing for deleting feedback off of an event.
+"""
+import logging
+from typing import Dict
+from fastapi.testclient import TestClient
+from requests.models import Response as HTTPResponse
+import models.users as user_models
+
+from app import app
+
+client = TestClient(app)
+"""
+Helper functions go outside the class
+"""
+
+
+def get_user_login_endpoint_url() -> str:  # pylint: disable=invalid-name
+    """
+    Returns the url string for the user login endpoint.
+
+    Might seem a little verbose and unnecessary but raw strings
+    are always trouble and take away lots of readability and flexibility.
+    """
+    return "/users/login"
+
+
+def get_user_login_json_data(user: user_models.User) -> Dict[str, str]:
+    """
+    Returns the url params needed to login the given user object
+    through the user login endpoint.
+    """
+    identifier = {"email": user.email}
+    password = user.password
+    return {"identifier": identifier, "password": password}
+
+
+def check_user_login_response_valid(response: HTTPResponse) -> bool:
+    """
+    Helper function that checks if status code is valid
+    """
+    try:
+        assert response.status_code == 200
```

Comment:
I'm having trouble figuring out how to assert there is a token in the response. The response is in the form of a `UserLoginResponse` object, yeah? So that means it will be json that includes strictly the jwt? But I'm not sure how to validate if what it returns _is_ a jwt, or do I just need to check the jwt field was actually filled out?

Reply:
We don't have the code that has the ability to check if a `jwt` is valid just yet, so just checking for existence is good enough at this point.

When that code gets into upstream we'll refactor this to check for `jwt` validity as well.